tests/test-lcfs.c

```diff
@@ -0,0 +1,32 @@
+#define _GNU_SOURCE
+
+#include "lcfs-writer.h"
+#include <assert.h>
+
+static inline void lcfs_node_unrefp(struct lcfs_node_s **nodep)
+{
+	if (*nodep != NULL) {
+		lcfs_node_unref(*nodep);
+		*nodep = NULL;
+	}
+}
+#define cleanup_node __attribute__((cleanup(lcfs_node_unrefp)))
+
+static void test_add_uninitialized_child(void)
+{
+	cleanup_node struct lcfs_node_s *node = lcfs_node_new();
+	lcfs_node_set_mode(node, S_IFDIR | 0755);
+	// libostree today does this pattern of creating an empty (uninitialized)
+	// child and passing it to lcfs_node_add_child first. Verify this
+	// continues to work for the forseeable future.
+	cleanup_node struct lcfs_node_s *child = lcfs_node_new();
+	int r = lcfs_node_add_child(node, child, "somechild");
+	assert(r == 0);
+	// Adding child took ownership
+	child = NULL;
+}
```

Comment:
Can you try writing an erofs here and making sure the validation error is flagged at that point?

Reply:
Sure done